pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -2166,8 +2169,19 @@ protected void processResults(ResultHandler handler, int flags) throws IOExcepti
           SimpleQuery currentQuery = executeData.query;
           Portal currentPortal = executeData.portal;
 
+          String nativeSql = currentQuery.getNativeQuery().nativeSql;
+          if (handler.getException() == null
+              && status.startsWith("ROLLBACK")
+              && ("COMMIT".equalsIgnoreCase(nativeSql)
+              || COMMIT_PATTERN.matcher(nativeSql).find()
+              && !ROLLBACK_PATTERN.matcher(nativeSql).find())) {
+            handler.handleError(
+                new PSQLException(
+                    GT.tr("The database returned ROLLBACK, so it looks like the transaction was aborted"),
```

Comment:
if and when the server ever fixes their code I doubt this will be the message we would send to the client. I think I'd like a generic "Error in Transaction ..." which does not mention anything about ROLLBACK

Reply:
It turns out we have `Throwable transactionFailCause`, so we already have the root cause of the transaction failure.